src/migrations/cluster-store/3.6.0-beta.1.ts

```diff
@@ -1,34 +1,57 @@
 // Move embedded kubeconfig into separate file and add reference to it to cluster settings
+// convert file path cluster icons to their base64 encoded versions
 
 import path from "path"
 import { app, remote } from "electron"
 import { migration } from "../migration-wrapper";
-import { ensureDirSync } from "fs-extra"
+import fse from "fs-extra"
 import { ClusterModel } from "../../common/cluster-store";
 import { loadConfig, saveConfigToAppFiles } from "../../common/kube-helpers";
 
 export default migration({
   version: "3.6.0-beta.1",
   run(store, printLog) {
-    const migratedClusters: ClusterModel[] = []
-    const storedClusters: ClusterModel[] = store.get("clusters");
     const kubeConfigBase = path.join((app || remote.app).getPath("userData"), "kubeconfigs")
+    const storedClusters: ClusterModel[] = store.get("clusters") || [];
 
-    if (!storedClusters) return;
-    ensureDirSync(kubeConfigBase);
+    if (!storedClusters?.length) return;
+    fse.ensureDirSync(kubeConfigBase);
 
     printLog("Number of clusters to migrate: ", storedClusters.length)
-    for (const cluster of storedClusters) {
-      try {
-        // take the embedded kubeconfig and dump it into a file
-        cluster.kubeConfigPath = saveConfigToAppFiles(cluster.id, cluster.kubeConfig)
-        cluster.contextName = loadConfig(cluster.kubeConfigPath).getCurrentContext();
-        delete cluster.kubeConfig;
-        migratedClusters.push(cluster)
-      } catch (error) {
-        printLog(`Failed to migrate Kubeconfig for cluster "${cluster.id}"`, error)
-      }
-    }
+    const migratedClusters = storedClusters
+      .map(cluster => {
+        /**
+         * migrate kubeconfig
+         */
+        try {
+          // take the embedded kubeconfig and dump it into a file
+          cluster.kubeConfigPath = saveConfigToAppFiles(cluster.id, cluster.kubeConfig)
+          cluster.contextName = loadConfig(cluster.kubeConfigPath).getCurrentContext();
+          delete cluster.kubeConfig;
+
+        } catch (error) {
+          printLog(`Failed to migrate Kubeconfig for cluster "${cluster.id}"`, error)
+          return undefined;
+        }
+
+        /**
+         * migrate cluster icon
+         */
+        try {
+          if (cluster.preferences.icon) {
+            const fileData = fse.readFileSync(cluster.preferences.icon);
```

Comment:
Is it was absolute path to icon, right?

Reply:
I think so, looking at https://github.com/lensapp/lens/blob/v3.5.0/src/main/cluster-manager.ts#L36